### ISSUE TITLE
Provide MyST-compatible output for `suggest`

### ISSUE THREAD

**Issue by laymonage** [bskinn/sphobjinv#300]
Hey, thanks for the great tool! It's very helpful.

**Is your feature request related to a problem? Please describe.**

Currently, the `suggest` command outputs the names in rST format. With the rise of MyST to write Sphinx docs in Markdown, it would be very useful if this tool could also output the suggested names in MyST format.

https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#cross-project-intersphinx-links

**Describe the solution you'd like**

Provide an argument for `suggest` to control the output format, e.g. `--format`, defaulting to `--format=rst`.

**Describe alternatives you've considered**

Don't implement it, and let something else convert rST-formatted name to MyST?

**Additional context**

From a quick look, we probably need to add a method e.g. `as_myst` in `SuperDataObj`, similar to `as_rst`:

https://github.com/bskinn/sphobjinv/blob/ff9151a6d00ad4693de1d4f541488f5c3ac04445/src/sphobjinv/data.py#L226-L236

and then maybe add an `objects_myst` property to `Inventory`:

https://github.com/bskinn/sphobjinv/blob/ff9151a6d00ad4693de1d4f541488f5c3ac04445/src/sphobjinv/inventory.py#L276-L291

then add a parameter to `suggest()` for the format:

https://github.com/bskinn/sphobjinv/blob/ff9151a6d00ad4693de1d4f541488f5c3ac04445/src/sphobjinv/inventory.py#L430-L438

and finally expose that in the parser...

https://github.com/bskinn/sphobjinv/blob/ff9151a6d00ad4693de1d4f541488f5c3ac04445/src/sphobjinv/cli/suggest.py#L41-L79

https://github.com/bskinn/sphobjinv/blob/ff9151a6d00ad4693de1d4f541488f5c3ac04445/src/sphobjinv/cli/parser.py#L37-L38

**Comment by bskinn** [comment]
Thanks, very glad it's been useful to you!

Interesting idea -- MyST is definitely here to stay, and odds are the next time I do anything major in Sphinx, I'll look carefully at doing that work in MyST.

The plan you lay out makes good sense to me, and I can't think of any implementation steps you've missed.

---

The one thing I'll push back on, lightly, is to ask: How valuable is emitting MyST formatting in addition to reST formatting, given that neither the reST currently emitted nor the MyST that'd be implemented for this would actually be a directly-usable cross-reference? (This being because the `objects.inv` and thus `sphobjinv` works with the long `role` names, which can't be used in cross-references.)

In other words -- MyST output wouldn't be directly copy-paste usable anyways, so isn't the current reST output good enough?

(That said, see #234, proposing a possible approach to having `sphobjinv` emit short-form `role` names, which  if it worked should allow both reST and MyST output to be copy-paste usable. It would _really_ be nice if this worked, but I'm a bit nervous that it'd turn out to be a major maintenance headache, trying to keep the directive conversion behavior compatible across Sphinx versions and over time.)
